[Native][ACR-1644]update ml_training api descriptions (#1516)
Samsung/tizen-docs: docs/application/native/guides/machine-learning/machine-learning-training.md

// Destroy after use.
ml_train_model_destroy(model);
```

A number of properties can be set at `ml_train_model_compile()` and `ml_train_model_run()` phase:

| Function                   | Key       | Value         | Description                            |
| -------------------------- | --------- | ------------- | -------------------------------------- |
| `ml_train_model_run()`     | epochs    | (integer)     | Determines epochs for the model        |
| &#xfeff;                   | save_path | (string)   | Model path to save parameters after a single epoch |

### Layer

Layer is a component that does actual computation while managing internal trainable parameters.
The following example shows how to create and add layer to the model:

```c
// Create layer
ml_train_layer_h layer;
ml_train_layer_create(&layer, ML_TRAIN_LAYER_TYPE_FC);

// Configure layer
ml_train_layer_set_property(layer, "unit=10", "activation=softmax", "bias_initializer=zeros", NULL);

// After adding the layer to model, you do not need to destroy layer since ownership is transferred to the model.
ml_train_model_add_layer(model, layer);
```

> There are 2 types of layers.

Author: Done

> One type includes commonly trainable weigths and another type does not includes.

Author: Done

> // Compile model. This freezes model and afterwards the model cannot be modified.

Reviewer: // Compile model - This freezes the model and afterward, the model cannot be modified.

Author: Done
-> // Compile model - This freezes the model. Afterwards the model cannot be modified.

ml_train_model_compile(model, "loss=cross", "batch_size=16", NULL);

> // run model

Reviewer: // Run model

Author: Done

ml_train_model_run(model, "epochs=2", NULL);

> /// Save, loading in below format supported since Tizen 6.5.

Reviewer: ```suggestion
// Save and load in the following format supported since Tizen 6.5:
```

Author: Done

ml_train_model_save(model, "model_weights.bin", ML_TRAIN_MODEL_FORMAT_BIN);
ml_train_model_save(model, "model.ini", ML_TRAIN_MODEL_FORMAT_INI);
ml_train_model_save(model, "model_and_weights.ini", ML_TRAIN_MODEL_FORMAT_INI_WITH_BIN);

// destroy after use
ml_train_model_destroy(model);
```

A number of properties can be set at `ml_train_model_compile()` and `ml_train_model_run()` phase:

| Function                   | Key       | Value         | Description                            |
| -------------------------- | --------- | ------------- | -------------------------------------- |

> | `ml_train_model_run()`     | epochs    | (integer)     | Determines epochs for the model        |

Reviewer: why is the value written in brackets?
(integer) and  (file_path)

Author: It should be (integer) and (string) <- updated.

The value in bracket means it's a family of type not the actual value.

IMO it's (integer, boolean, string, floats) are quite common so wouldn't need additional explanation

| &#xfeff;                   | save_path | (file_path)   | Model path to save parameters after a single epoch |

### Layer

Layer is a component that does actual computation while managing internal trainable parameters.

> This is how to create and add layer to the model.

Reviewer: ```suggestion
The following example shows how to create and add layer to the model:
```

Author: Done

```c
// Create layer
ml_train_layer_h layer;
ml_train_layer_create(&layer, ML_TRAIN_LAYER_TYPE_FC);

> // configure layer

Reviewer: // Configure layer

Author: Done

ml_train_layer_set_property(layer, "unit=10", "activation=softmax", "bias_initializer=zeros", NULL);

> // after adding the layer to model,

Reviewer: ```suggestion
// After adding the layer to model, you do not need to destroy the layer as the ownership is transferred to the model.
```

// you do not need to destroy layer since ownership is transferred to the model.
ml_train_model_add_layer(model, layer);
```

There are 2 types of layers.
One type includes commonly trainable weigths and another type does not includes.

> The following are the available properties for each layer type which include commonly trainable weights.

Reviewer: ```suggestion
The following are the available properties for each layer type which include commonly trainable weights;
```

Author: Changed to colon.

Type | Key | Value | Default value | Description

> ---------- | --- | ----- | ----------- | -----------

Reviewer: Some of the values are in brackets some are not in, any specific reason?
Recommend to follow consistency.

Author: discussed in https://github.com/Samsung/tizen-docs/pull/1516#discussion_r716361917

&#xfeff;                                                     |                             | tanh                        |                         | Hyperbolic tangent
&#xfeff;                                                     |                             | sigmoid                     |                         | Sigmoid function
&#xfeff;                                                     |                             | relu                        |                         | Relu function
&#xfeff;                                                     |                             | softmax                     |                         | Softmax function
&#xfeff;                                                     | return_sequences            | (boolean)                   | false                   | Return only the last output if true, else return full output
&#xfeff;                                                     | dropout                     | (float)                     | 0                       | Dropout rate

> The following are the available properties for each layer type which doesn't include (weight_initializer, bias_initializer, weight_regularizer, weight_regularizer_constant) properties.

Reviewer: ```suggestion
The following are the available properties for each layer type which does not include (`weight_initializer`, `bias_initializer,` `weight_regularizer`, `weight_regularizer_constant`) properties;
```

Author: Done

There are two ways to create a dataset. One is from a file, and the other one is from a callback.
In either case, you need to provide streams of tensor data and arrays of values representing the label, usually one-hot-encoded.

#### Sample, batch and epoch

This section explains the following three concepts: sample, batch and epoch.

Let's assume a given model requires three inputs and two labels.

1. Sample

A sample denotes a single pair of inputs/labels.

```
[input_1][input_2][input_3][label_1][label_2]
```

2. Batch

> A batch is a bundle of samples which will be fed to a single iteration. If a batch is of size 5, the batch layout, where `/*` denotes nth sample will be:

Author: Done

```
[[input_1/1][input_1/2][input_1/3][input_1/4][input_1/5]],
[[input_2/1][input_2/2][input_2/3][input_2/4][input_2/5]],
[[input_3/1][input_3/2][input_3/3][input_3/4][input_3/5]],
[[label_1/1][label_1/2][label_1/3][label_1/4][label_1/5]],
[[label_2/1][label_2/2][label_2/3][label_2/4][label_2/5]],
```

3. Epoch

An epoch refers to a full, exhaustive visit to a dataset.

> So, if you consider [Cifar-100](https://www.cs.toronto.edu/~kriz/cifar.html#The%20CIFAR-100%20dataset) dataset. When using 100% of training set, a sigle training epoch will be visiting 50,000 samples. If batch size is 100, it will consists of 500 batches(or iterations).

Author: Could you verify below alternative?

`When using the full training set, a single training epoch will contain 50,000 samples. If batch size is 100, it will be of 500 batches(or iterations).`

I think we can't use the word `can` it's not about `it cant be`, but it is sort of the description of the fact.

There are two ways to create a dataset. One is from a file, and the other one is from a callback.
In either case, you need to provide streams of tensor data and arrays of values representing the label, usually one-hot-encoded.

#### Sample, batch and epoch

> There are three confusing concepts need clarification: sample, batch and epoch.

Reviewer: ```suggestion
This section explains the following three concepts: sample, batch and epoch.
```

Author: Done

`ml_train_model_add_layer()`, `ml_train_set_optimizer()`, and `ml_train_set_dataset()` transfers ownership to the model.
`layers`, `optimizers` and `dataset` that belongs to the `model` are also deleted.

## Use the trained model for inference

> [!NOTE]

> > The feature explained in this section is supported since Tizen 6.5

Author: Done

> > The feature explained in this section is supported since Tizen 6.5

Author: Done

The trained model can be used for inference with [Machine Learning Inference API](machine-learning-inference.md).

> Ensure that the INI file contains the correct weight file in `save_path` in `[Model]` section. The valid INI file can be made from `ml_train_model_h` with `ml_train_model_save()` if you have constructed the model with the provided api.

Author: Done

`ml_train_model_add_layer()`, `ml_train_set_optimizer()`, and `ml_train_set_dataset()` transfers ownership to the model.
`layers`, `optimizers` and `dataset` that belongs to the `model` are also deleted.

> ## Use the trained model for inference (since Tizen 6.5)

Reviewer: Please include since Tizen 6.5 as a note.

Author: Done

    ...

    ml_single_open (&single, "model_file.ini", in_info, out_info, ML_NNFW_TYPE_NNTR_INF, ML_NNFW_HW_ANY);
  ```

> You can use the trained model with with pipeline API as follows:

Author: Done
